src/serializers/type_serializers/definitions.rs

```diff
@@ -93,8 +94,12 @@ impl TypeSerializer for DefinitionRefSerializer {
     ) -> PyResult<PyObject> {
         self.definition.read(|comb_serializer| {
             let comb_serializer = comb_serializer.unwrap();
-            let mut guard = extra.recursion_guard(value, self.definition.id())?;
-            comb_serializer.to_python(value, include, exclude, guard.state())
+            if extra.duck_typing_ser_mode != DuckTypingSerMode::Inferred {
```

Comment:
so what I found is that the `outer serializer` call to `rescursive serializer` but also put the `self.definition.id()` in the stack already. So when the `rescursive serializer` start to eval, the root model of it have the same `self.definition.id()` result in raise duplicated id error. 

Checking for `DuckTypingSerMode::Inferred` is because that what I saw as a flag when the `rescursive serializer` started. I could be wrong so very open for suggestions !

Reply:
This change (to not introduce recursion guards) is the cause of the segfaults in the tests in CI.

Reply:
have fixed the issue!